lib/ClangImporter/ImportDecl.cpp

```diff
@@ -4590,12 +4590,37 @@ namespace {
       auto introducer = Impl.shouldImportGlobalAsLet(decl->getType())
                         ? VarDecl::Introducer::Let
                         : VarDecl::Introducer::Var;
-      auto result = Impl.createDeclWithClangNode<VarDecl>(
-          decl, importer::convertClangAccess(decl->getAccess()),
-          /*IsStatic*/ isStatic, introducer,
-          Impl.importSourceLoc(decl->getLocation()), name, dc);
-      result->setIsObjC(false);
-      result->setIsDynamic(false);
+
+      ValueDecl *result = nullptr;
+
+      // If the variable is const (we're importing it as a let), and has an
+      // initializer, then ask Clang for its constant value and synthesize a
+      // getter with that value.
+      if (introducer == VarDecl::Introducer::Let && decl->hasInit()) {
+        auto val = decl->evaluateValue();
+        // For now, only import integer and float constants. If in the future
+        // SwiftDeclSynthesizer::createConstant becomes able to import more
+        // types, we can lift this restriction.
+        if (val && (val->isFloat() || val->isInt())) {
```

Comment:
Do we have a test case where `val` would be null here? Do we have one where the type tests fail?

Reply:
Added the tests. My reading of `evaluateValue()` is that the notes are discarded (unlike e.g. `checkForConstantInitialization()` which Clang seems to use to actually surface the notes to the user). That seems like a good conservative default here (i.e. if the value doesn't constant fold, we silently fall back to importing as an external declaration)?